modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsConfigurationManager.java

```diff
@@ -0,0 +1,518 @@
+            }
+            catch (IgniteCheckedException e) {
+                log.warning("Unexpected exception on check local statistic on start", e);
+            }
+        });
+    }
+
+    /** */
+    public void updateStatistics(List<StatisticsTarget> targets) {
+        if (log.isDebugEnabled())
+            log.debug("Update statistics [targets=" + targets + ']');
+
+        for (StatisticsTarget target : targets) {
+            GridH2Table tbl = schemaMgr.dataTable(target.schema(), target.obj());
+
+            validate(target, tbl);
+
+            Column[] cols = IgniteStatisticsHelper.filterColumns(tbl.getColumns(), Arrays.asList(target.columns()));
+
+            StatisticsColumnConfiguration[] colCfgs = Arrays.stream(cols)
+                .map(c -> new StatisticsColumnConfiguration(c.getName()))
+                .collect(Collectors.toList())
+                .toArray(new StatisticsColumnConfiguration[cols.length]);
+
+            updateObjectStatisticConfiguration(
+                new StatisticsObjectConfiguration(
+                    target.key(),
+                    colCfgs
+                )
+            );
+        }
+    }
+
+    /** */
+    public void dropStatistics(List<StatisticsTarget> targets) {
+        if (log.isDebugEnabled())
+            log.debug("Drop statistics [targets=" + targets + ']');
+
+        for (StatisticsTarget target : targets) {
+            String key = key2String(target.key());
+
+            try {
+                while (true) {
+                    StatisticsObjectConfiguration oldCfg = distrMetaStorage.read(key);
+
+                    if (oldCfg == null)
+                        break;
+
+                    StatisticsColumnConfiguration[] newColsCfg = filterColumns(
+                        oldCfg,
+                        Arrays.stream(target.columns()).collect(Collectors.toSet())
+                    );
+
+                    StatisticsObjectConfiguration newCfg = new StatisticsObjectConfiguration(
+                        target.key(),
+                        newColsCfg,
+                        oldCfg.version()
+                    );
+
+                    if (distrMetaStorage.compareAndSet(key, oldCfg, newCfg))
+                        break;
+                }
+            }
+            catch (IgniteCheckedException ex) {
+                throw new IgniteSQLException("Error on get or update statistic schema", IgniteQueryErrorCode.UNKNOWN, ex);
+            }
+        }
+    }
+
+    /** */
+    private StatisticsColumnConfiguration[] filterColumns(
+        StatisticsObjectConfiguration oldCfg,
+        Set<String> colToRemove
+    ) {
+        // Drop all columns
+        if (colToRemove.isEmpty())
+            return EMPTY_COLUMN_CFGS_ARR;
+
+        Set<StatisticsColumnConfiguration> cols = Arrays.stream(oldCfg.columns())
+            .filter(c -> !colToRemove.contains(c.name())).collect(Collectors.toSet());
+
+        // All not-hidden fields are dropped -> Drop all columns
```

Comment:
When we need _KEY or _VALUE covered by statistics? Can we just throw an error while one trying to analyze them?